ml_peg/app/utils/utils.py

```diff
@@ -3,9 +3,14 @@
 from __future__ import annotations
 
 from collections.abc import Mapping, MutableMapping, Sequence
```

Comment:
This wasn't newly introduced, but how many of the type hints are actually only `Mapping`s, and not always a `dict` in reality?

It's not something I've encountered a huge amount, but I'm pretty sure `Mapping` as an abstract base class doesn't require the implementation of quite a few things that we may depend on, such as `pop`, `update`, `__setitem__` (I think `MutableMapping` does), `__delitem__`, or `setdefault`.

I think in most cases these probably should just be `dict`0s